kernel: Remove args, settings, chainparams, chainparamsbase from kernel library

sedited opened the pull request:
This pull request is part of the `libbitcoinkernel` project https://github.com/bitcoin/bitcoin/issues/27587 https://github.com/bitcoin/bitcoin/projects/18 and more specifically its "Step 2: Decouple most non-consensus code from libbitcoinkernel".

---

This completes the removal of the node's chainparams, chainparamsbase, args and settings files and their respective classes from the kernel library. This is the last pull request in a long series working towards decoupling the `ArgsManager` and the `gArgs` global from kernel code. These prior pull requests are: https://github.com/bitcoin/bitcoin/pull/26177 https://github.com/bitcoin/bitcoin/pull/27125 https://github.com/bitcoin/bitcoin/pull/25527 https://github.com/bitcoin/bitcoin/pull/25487 https://github.com/bitcoin/bitcoin/pull/25290

sedited committed:
refactor: Add stop_at_height option in ChainstateManager

Remove access to the global gArgs for the stopatheight argument and
replace it by adding a field to the existing ChainstateManager Options
struct.

This should eventually allow users of the ChainstateManager to not rely
on the global gArgs and instead pass in their own options.

sedited committed:
refactor: Add path argument to FindSnapshotChainstateDir

Remove access to the global gArgs for getting the directory in
utxo_snapshot.

This is done in the context of the libbitcoinkernel project, wherein
reliance of libbitcoinkernel code on the global gArgs is incrementally
removed.

sedited committed:
refactor: Remove gArgs access from validation.cpp

This is done in the context of the libbitcoinkernel project, wherein
reliance of libbitcoinkernel code on the global gArgs is incrementally
removed.

sedited committed:
kernel: Remove chainparams, chainparamsbase, args, settings from kernel library

fanquake commented:
```bash
src/common/settings.h seems to be missing the expected include guard:
  #ifndef BITCOIN_COMMON_SETTINGS_H
  #define BITCOIN_COMMON_SETTINGS_H
  ...
  #endif // BITCOIN_COMMON_SETTINGS_H
```

hebasto commented:
Concept ACK.

sedited committed:
move-only: Move settings to the common library

The background of this commit is an ongoing effort to decouple the
libbitcoinkernel library from code that is not strictly required by it.
The settings code belongs into the common library and namespace, since
the kernel library should not depend on it. See doc/design/libraries.md
for more information on this rationale.

Changing the namespace of the moved functions is scripted in the
following commit.

sedited committed:
scripted-diff: move settings to common namespace

-BEGIN VERIFY SCRIPT-
sed -i 's/namespace\ util/namespace\ common/g' src/common/settings.cpp src/common/settings.h
sed -i 's/util\:\:GetSetting/common\:\:GetSetting/g' $( git grep -l 'util\:\:GetSetting')
sed -i 's/util\:\:Setting/common\:\:Setting/g' $( git grep -l 'util\:\:Setting')
sed -i 's/util\:\:FindKey/common\:\:FindKey/g' $( git grep -l 'util\:\:FindKey')
sed -i 's/util\:\:ReadSettings/common\:\:ReadSettings/g' $( git grep -l 'util\:\:ReadSettings')
sed -i 's/util\:\:WriteSettings/common\:\:WriteSettings/g' $( git grep -l 'util\:\:WriteSettings')
-END VERIFY SCRIPT-

hebasto commented on src/Makefile.am:
```diff
@@ -912,12 +912,8 @@ libbitcoinkernel_la_SOURCES = \
   clientversion.cpp \
   coins.cpp \
-  common/args.cpp \
-  common/config.cpp \
```
Side note: Removing of `common/config.cpp` is correct but not directly related to this PR changes as it can be done even on the current master branch.

hebasto reviewed (approved):
ACK db77f87c6365cb5f414036d6bfb1a12705772028, I have reviewed the code and it looks OK.

ryanofsky commented on src/test/validation_chainstatemanager_tests.cpp:
```diff
@@ -184,7 +184,7 @@ struct SnapshotTestSetup : TestChain100Setup {
             LOCK(::cs_main);
             BOOST_CHECK(!chainman.IsSnapshotValidated());
-            BOOST_CHECK(!node::FindSnapshotChainstateDir());
+            BOOST_CHECK(!node::FindSnapshotChainstateDir(m_args.GetDataDirNet()));
```
In commit "refactor: Add path argument to FindSnapshotChainstateDir" (8789b11114b4bd6c7ee727dffbc75a6bdf20dd27)

Throughout this file, it would make tests clearer and reliant on details of test setup if the new
`m_args.GetDataDirNet()` calls were replaced with `chainman.m_options.datadir`

fanquake replied:
@TheCharlatan did you want to open a followup for this?

TheCharlatan replied:
Yes.

ryanofsky reviewed (approved):
Code review ACK db77f87c6365cb5f414036d6bfb1a12705772028. Looks great!

I left a suggestion to clean up a unit test in a possible followup, but I will just go ahead and merge this now if tests pass locally.

ryanofsky reviewed (approved):
In commit "scripted-diff: move settings to common namespace" (db77f87c6365cb5f414036d6bfb1a12705772028)

Not important but I noticed the scripted diff is escaping space and colon characters. I think there is not actually a need for all those backslashes, and they don't do anything

EDIT: just noticed Marco left the same comment. Great minds, I guess
